pkgs/desktops/deepin/artwork/dde-account-faces/default.nix

```
@@ -0,0 +1,31 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "dde-account-faces";
+  version = "1.0.12.1";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-NWjR8qxWi2IrcP0cSF+lLxBJ/GrVpk1BfTjVH0ytinY=";
+  };
+
+  # It should be installed to /var，but this can't be done directly on nixos, so move to $out/share
+  # we need patch dde-control-center / dde-daemon also
+  makeFlags = [
+    "DESTDIR=$(out)/share"
+    "PREFIX=/"
+  ];
```

I would just use

```suggestion
  makeFlags = [
    # Package uses incorrect variable name (instead of standard localstatedir).
    "PREFIX=${placeholder "out"}/var"
  ];
```

and then map it to `/var` in the NixOS module using `systemd.tmpfiles`.

---

Good idea, i'll test it later

---

if I map `/var/lib/AccountsService/icons/`, this directory is not writable，Users cannot configure custom avatars

---

The `C` directive of `tmpfiles.d` can be used to copy the contents only if the destination directory is empty. Effectively initializing `/var/lib/AccountsService/icons/` with a default set of contents, while keeping the directory writable.

---

LGTM https://github.com/linuxdeepin/dde-nixos/commit/406dad3bc27df8dfb744585d119378bdebea1ab6